aptos-move/framework/src/module_metadata.rs

```diff
@@ -148,31 +149,64 @@ impl KnownAttribute {
     }
 }
 
+const METADATA_CACHE_SIZE: usize = 1024;
+
+thread_local! {
+    static V1_METADATA_CACHE: RefCell<LruCache<Vec<u8>, Option<Arc<RuntimeModuleMetadataV1>>>> = RefCell::new(LruCache::new(METADATA_CACHE_SIZE));
```

Comment:
As discussed on #9320, this is likely not ideal (since the key can be kb-ish), but better than we have right now! Thanks for doing this as a temporary fix.

Reply:
Yeah it would be nice if modules can live across blocks and carry the metadata themselves.